resolver/src/resolver_future.rs

```diff
@@ -243,6 +243,11 @@ impl ResolverFuture {
     /// # Arguments
     /// * `host` - string hostname, if this is an invalid hostname, an error will be returned.
     pub fn lookup_ip(&self, host: &str) -> LookupIpFuture {
+        // if host is a ip address, return directly.
+        if let Some(addr) = parse_literal_ip(host) {
```

Comment:
there's a `warn!` in the `parse_literal_ip` method in the case of errors. Given that this is really intended for host names and not just IPs, can we move that `warn!` to `read_hosts_conf`?

Reply:
fixed